### ISSUE TITLE
Seed and sample behaviour in Tuple space

### ISSUE THREAD

**Issue by kevinmcareavey** [openai/gym#1864]
If I create a Tuple space containing Discrete spaces, fix the seed, and then generate a sample, all elements of the sampled tuple are the same.  For example, (0, 0) or (4, 4).  Conversely, if I create a MultiDiscrete space, fix the seed, and then generate a sample, elements of the sampled tuple are not necessarily the same.  For example, (0, 4) or (1, 3).

This is especially problematic when a Tuple space contains Tuple spaces (i.e. where MultiDiscrete is not an adequate replacement):

```python
from gym.spaces import Tuple, MultiDiscrete, Discrete

my_seed = 0

# Test 1

print("Test 1:")

a1 = Tuple([Tuple([Discrete(5) for _ in range(2)]) for _ in range(2)])
b1 = Tuple([MultiDiscrete([5 for _ in range(2)]) for _ in range(2)])

a1.seed(my_seed)
b1.seed(my_seed)

for _ in range(5):
    print(a1.sample(), b1.sample())

# Test 2

print("Test 2:")

a2 = Tuple([Tuple([Discrete(5) for _ in range(2)]) for _ in range(2)])
b2 = Tuple([MultiDiscrete([5 for _ in range(2)]) for _ in range(2)])

for a2_space, b2_space in zip(a2, b2):
    my_seed += 1
    a2_space.seed(my_seed)
    b2_space.seed(my_seed)

for _ in range(5):
    print(a2.sample(), b2.sample())
```
```
Test 1:
((0, 0), (0, 0)) (array([0, 4]), array([0, 4]))
((3, 3), (3, 3)) (array([3, 1]), array([3, 1]))
((4, 4), (4, 4)) (array([0, 3]), array([0, 3]))
((3, 3), (3, 3)) (array([1, 3]), array([1, 3]))
((3, 3), (3, 3)) (array([0, 2]), array([0, 2]))
Test 2:
((0, 0), (4, 4)) (array([4, 2]), array([4, 3]))
((4, 4), (0, 0)) (array([0, 0]), array([3, 3]))
((3, 3), (0, 0)) (array([3, 0]), array([4, 2]))
((2, 2), (4, 4)) (array([0, 2]), array([1, 2]))
((4, 4), (1, 1)) (array([3, 2]), array([2, 2]))
```

Is this expected/desirable behaviour for Tuple or am I using it incorrectly?  I would have expected Tuple to behave for arbitrary tuples as MultiDiscrete behaves for a single tuple.

**Comment by kevinmcareavey** [comment]
This issue appears to be due to the `Tuple` contructor, which resets the seed of each tuple element to `None`:

https://github.com/openai/gym/blob/b2727d6fd396f7fe2b19a541ef3a6ea47559cdb3/gym/spaces/tuple.py#L16

https://github.com/openai/gym/blob/b2727d6fd396f7fe2b19a541ef3a6ea47559cdb3/gym/spaces/space.py#L14

https://github.com/openai/gym/blob/b2727d6fd396f7fe2b19a541ef3a6ea47559cdb3/gym/spaces/tuple.py#L18-L19

A possible workaround is to recursively reset `np_random`:
```python
def set_np_random(space, np_random):
    space.np_random = np_random
    if isinstance(space, spaces.Tuple):
        for element in space:
            set_np_random(element, np_random)```

**Comment by kevinmcareavey** [comment]
This issue is not resolved in 0.17.3 but the previously suggested workaround is now broken, i.e.:
> ```python
> def set_np_random(space, np_random):
>     space.np_random = np_random
>     if isinstance(space, spaces.Tuple):
>         for element in space:
>             set_np_random(element, np_random)
> ```
